sql/src/main/java/io/crate/metadata/settings/session/SessionSetting.java

```diff
@@ -64,4 +77,12 @@ public void apply(Row parameters, List<Expression> expressions, SessionContext s
     public String getValue(SessionSettings sessionSettings) {
         return getter.apply(sessionSettings);
     }
+
+    public String defaultValue() {
+        return defaultValue.get();
+    }
+
+    public Map<Property, String> properties() {
```

Comment:
What's the reasoning for this `Property` indirection? Why not make them first class explicit properties on `SessionSettings` ?

Reply:
If we need to add some additional properties, e.g. unit, group, etc. for new or existing settings, it might be a bit easier to extend and to reuse in the wrapper class. That basically was the reasoning, shall we then convert them to SessionSetting fields since there is only two of them now?

Reply:
I don't think that extending an enum and a map with values is easier than adding new properties to a class. But using a map has the downside that we don't enforce them to be present.

So yep, I'd go with having concrete attributes/properties on the SessionSetting class